jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/util/WorkItemHeaderUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.core.util;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.jbpm.process.workitem.core.util.WorkItemHeaderInfo.Builder;
+import org.kie.api.runtime.process.WorkItem;
+
+public class WorkItemHeaderUtils {
+
+    private WorkItemHeaderUtils() {}
+
+    private static final String PREFIX = "HEADER_";
+
+    public static Collection<WorkItemHeaderInfo> getHeaderInfo(WorkItem workItem) {
+        Map<String, WorkItemHeaderInfo.Builder> map = new HashMap<>();
+        for (Entry<String, Object> param : workItem.getParameters().entrySet()) {
+            if (param.getKey().toUpperCase().startsWith(PREFIX)) {
+                String name = param.getKey().substring(PREFIX.length());
+                String paramKey = null;
+                int indexOf = name.lastIndexOf("_");
+                if (indexOf != -1) {
+                    paramKey = name.substring(indexOf + 1);
+                    name = name.substring(0, indexOf);
+                }
+                Builder builder = map.computeIfAbsent(name, Builder::of);
+                if (paramKey != null) {
```

Comment:
Maybe it could be also checked if paramKey is also not empty

Reply:
hmm, if a user has put an empty string there (if not this will be null) I think the user should have precedence